Newer libc versions (0.2.113 or later) don't compile with Rust-GPU's rust-toolchain.

<!-- Thank you for filing a bug report! 🐛 -->

### Expected Behaviour
Following the steps described in the [docs](https://embarkstudios.github.io/rust-gpu/book/writing-shader-crates.html) should work without errors in the supported platforms.

### Example & Steps To Reproduce

1. Create a new blanket rust project with cargo new
2. Copy the rust tool-chain file from git onto the root of the project
3. Add ```[build-dependencies]
spirv-builder = { git = "https://github.com/EmbarkStudios/rust-gpu" }``` to the toml file.
4. Create a `build.rs` file, copy the snippet from the docs and replace the path to the shader and the target with the relevant strings.

## System Info

 - Rust: rust version rustc 1.62.0-nightly (1f7fb6413 2022-04-10)
 - OS: Ubuntu 22.04
 - GPU: Nvidia GTX 1070
 - SPIR-V: vulkan


## Backtrace
```
error[E0658]: `const extern fn` definitions are unstable
    --> /home/makogan/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/macros.rs:188:17
     |
188  | /                 pub $($constness)* unsafe extern fn $i($($arg: $argty),*
189  | |                 ) -> $ret {
190  | |                     $($body);*
191  | |                 }
     | |_________________^
     |
    ::: /home/makogan/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/unix/linux_like/mod.rs:1499:1
     |
1499 | / f! {
1500 | |     pub fn CMSG_FIRSTHDR(mhdr: *const msghdr) -> *mut cmsghdr {
1501 | |         if (*mhdr).msg_controllen as usize >= ::mem::size_of::<cmsghdr>() {
1502 | |             (*mhdr).msg_control as *mut cmsghdr
...    |
1545 | |     }
1546 | | }
     | |_- in this macro invocation
     |
     = note: see issue #64926 <https://github.com/rust-lang/rust/issues/64926> for more information
     = help: add `#![feature(const_extern_fn)]` to the crate attributes to enable
     = note: this error originates in the macro `f` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: `const extern fn` definitions are unstable
    --> /home/makogan/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/macros.rs:203:17
     |
203  | /                 pub $($constness)* extern fn $i($($arg: $argty),*
204  | |                 ) -> $ret {
205  | |                     $($body);*
206  | |                 }
     | |_________________^
     |
    ::: /home/makogan/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/unix/linux_like/mod.rs:1548:1
     |
1548 | / safe_f! {
1549 | |     pub fn SIGRTMAX() -> ::c_int {
1550 | |         unsafe { __libc_current_sigrtmax() }
1551 | |     }
...    |
1615 | |     }
1616 | | }
     | |_- in this macro invocation
     |
     = note: see issue #64926 <https://github.com/rust-lang/rust/issues/64926> for more information
     = help: add `#![feature(const_extern_fn)]` to the crate attributes to enable
     = note: this error originates in the macro `safe_f` (in Nightly builds, run with -Z macro-backtrace for more info)
```

The example that you have provided does not compile on my setup:
`cargo check`:
```rust
   Compiling libc v0.2.131
   Compiling ahash v0.7.6
   Compiling sanitize-filename v0.4.0
   Compiling quote v1.0.21
   Compiling num-traits v0.2.15
   Compiling indexmap v1.9.1
error[E0658]: `const extern fn` definitions are unstable
   --> /Users/wiktorjurkiewicz/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/macros.rs:203:17
    |
203 | /                 pub $($constness)* extern fn $i($($arg: $argty),*
204 | |                 ) -> $ret {
205 | |                     $($body);*
206 | |                 }
    | |_________________^
    |
   ::: /Users/wiktorjurkiewicz/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/unix/bsd/mod.rs:572:1
    |
572 | / safe_f! {
573 | |     pub {const} fn WTERMSIG(status: ::c_int) -> ::c_int {
574 | |         status & 0o177
575 | |     }
...   |
591 | |     }
592 | | }
    | |_- in this macro invocation
    |
    = note: see issue #64926 <https://github.com/rust-lang/rust/issues/64926> for more information
    = help: add `#![feature(const_extern_fn)]` to the crate attributes to enable
    = note: this error originates in the macro `safe_f` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: `const extern fn` definitions are unstable
    --> /Users/wiktorjurkiewicz/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/macros.rs:188:17
     |
188  | /                 pub $($constness)* unsafe extern fn $i($($arg: $argty),*
189  | |                 ) -> $ret {
190  | |                     $($body);*
191  | |                 }
     | |_________________^
     |
    ::: /Users/wiktorjurkiewicz/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/unix/bsd/apple/mod.rs:4758:1
     |
4758 | / f! {
4759 | |     pub fn CMSG_NXTHDR(mhdr: *const ::msghdr,
4760 | |                        cmsg: *const ::cmsghdr) -> *mut ::cmsghdr {
4761 | |         if cmsg.is_null() {
...    |
4793 | |     }
4794 | | }
     | |_- in this macro invocation
     |
     = note: see issue #64926 <https://github.com/rust-lang/rust/issues/64926> for more information
     = help: add `#![feature(const_extern_fn)]` to the crate attributes to enable
     = note: this error originates in the macro `f` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: `const extern fn` definitions are unstable
    --> /Users/wiktorjurkiewicz/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/macros.rs:203:17
     |
203  | /                 pub $($constness)* extern fn $i($($arg: $argty),*
204  | |                 ) -> $ret {
205  | |                     $($body);*
206  | |                 }
     | |_________________^
     |
    ::: /Users/wiktorjurkiewicz/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.131/src/unix/bsd/apple/mod.rs:4796:1
     |
4796 | / safe_f! {
4797 | |     pub {const} fn WSTOPSIG(status: ::c_int) -> ::c_int {
4798 | |         status >> 8
4799 | |     }
...    |
4815 | |     }
4816 | | }
     | |_- in this macro invocation
     |
     = note: see issue #64926 <https://github.com/rust-lang/rust/issues/64926> for more information
     = help: add `#![feature(const_extern_fn)]` to the crate attributes to enable
     = note: this error originates in the macro `safe_f` (in Nightly builds, run with -Z macro-backtrace for more info)

   Compiling syn v1.0.99
For more information about this error, try `rustc --explain E0658`.
error: could not compile `libc` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -94,8 +94,7 @@ jobs:
           echo "::endgroup::"
 
           echo "::group::Install cargo-apk"
-          echo "FIXME(eddyb) working around libc 0.2.127 breakage by using stable Rust!"
-          cargo +stable install cargo-apk
+          cargo install cargo-apk
           echo "::endgroup::"
 
           echo "::group::Add aarch64-linux-android target"

--- a/crates/rustc_codegen_spirv/src/codegen_cx/mod.rs
+++ b/crates/rustc_codegen_spirv/src/codegen_cx/mod.rs
@@ -671,11 +671,11 @@ impl<'tcx> CoverageInfoMethods<'tcx> for CodegenCx<'tcx> {
     }
 }
 
-impl<'tcx> AsmMethods for CodegenCx<'tcx> {
+impl<'tcx> AsmMethods<'tcx> for CodegenCx<'tcx> {
     fn codegen_global_asm(
         &self,
         _template: &[InlineAsmTemplatePiece],
-        _operands: &[GlobalAsmOperandRef],
+        _operands: &[GlobalAsmOperandRef<'tcx>],
         _options: InlineAsmOptions,
         _line_spans: &[Span],
     ) {

--- a/rust-toolchain
+++ b/rust-toolchain
@@ -5,5 +5,5 @@
 # to the user in the error, instead of "error: invalid channel name '[toolchain]'".
 
 [toolchain]
-channel = "nightly-2022-08-15"
+channel = "nightly-2022-04-18"
 components = ["rust-src", "rustc-dev", "llvm-tools-preview"]

--- a/rust-toolchain
+++ b/rust-toolchain
@@ -5,5 +5,5 @@
 # to the user in the error, instead of "error: invalid channel name '[toolchain]'".
 
 [toolchain]
-channel = "nightly-2022-04-11"
+channel = "nightly-2022-08-15"
 components = ["rust-src", "rustc-dev", "llvm-tools-preview"]

--- a/tests/ui/arch/debug_printf_type_checking.stderr
+++ b/tests/ui/arch/debug_printf_type_checking.stderr
@@ -59,26 +59,42 @@ error: Unrecognised format specifier: 'r'
    |                       ^^^^
 
 error[E0308]: mismatched types
-  --> $DIR/debug_printf_type_checking.rs:20:29
-   |
-20 |         debug_printf!("%f", 11_u32);
-   |                             ^^^^^^ expected `f32`, found `u32`
-   |
+   --> $DIR/debug_printf_type_checking.rs:20:29
+    |
+20  |         debug_printf!("%f", 11_u32);
+    |         --------------------^^^^^^-
+    |         |                   |
+    |         |                   expected `f32`, found `u32`
+    |         arguments to this function are incorrect
+    |
+note: function defined here
+   --> $SPIRV_STD_SRC/lib.rs:138:8
+    |
+138 | pub fn debug_printf_assert_is_type<T>(ty: T) -> T {
+    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 help: change the type of the numeric literal from `u32` to `f32`
-   |
-20 |         debug_printf!("%f", 11_f32);
-   |                                ~~~
+    |
+20  |         debug_printf!("%f", 11_f32);
+    |                                ~~~
 
 error[E0308]: mismatched types
-  --> $DIR/debug_printf_type_checking.rs:21:29
-   |
-21 |         debug_printf!("%u", 11.0_f32);
-   |                             ^^^^^^^^ expected `u32`, found `f32`
-   |
+   --> $DIR/debug_printf_type_checking.rs:21:29
+    |
+21  |         debug_printf!("%u", 11.0_f32);
+    |         --------------------^^^^^^^^-
+    |         |                   |
+    |         |                   expected `u32`, found `f32`
+    |         arguments to this function are incorrect
+    |
+note: function defined here
+   --> $SPIRV_STD_SRC/lib.rs:138:8
+    |
+138 | pub fn debug_printf_assert_is_type<T>(ty: T) -> T {
+    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 help: change the type of the numeric literal from `f32` to `u32`
-   |
-21 |         debug_printf!("%u", 11u32);
-   |                               ~~~
+    |
+21  |         debug_printf!("%u", 11u32);
+    |                               ~~~
 
 error[E0277]: the trait bound `{float}: Vector<f32, 2_usize>` is not satisfied
    --> $DIR/debug_printf_type_checking.rs:22:31
@@ -106,10 +122,19 @@ note: required by a bound in `debug_printf_assert_is_vector`
     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `debug_printf_assert_is_vector`
 
 error[E0308]: mismatched types
-  --> $DIR/debug_printf_type_checking.rs:23:29
-   |
-23 |         debug_printf!("%f", Vec2::splat(33.3));
-   |                             ^^^^^^^^^^^^^^^^^ expected `f32`, found struct `Vec2`
+   --> $DIR/debug_printf_type_checking.rs:23:29
+    |
+23  |         debug_printf!("%f", Vec2::splat(33.3));
+    |         --------------------^^^^^^^^^^^^^^^^^-
+    |         |                   |
+    |         |                   expected `f32`, found struct `Vec2`
+    |         arguments to this function are incorrect
+    |
+note: function defined here
+   --> $SPIRV_STD_SRC/lib.rs:138:8
+    |
+138 | pub fn debug_printf_assert_is_type<T>(ty: T) -> T {
+    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error: aborting due to 14 previous errors
 

